Dashboard

klucar opened the pull request:
Completely new dashboard single page application for Myriad. 

Uses Facebook's React component system to build and update user interface completely on the client side.

Uses npm for dependencies and gulp to compile js artifacts together.

adam-mesos commented:
Is this apache-licensed, or friendly with the Apache license?

klucar commented:
From what I can tell, yes it is Apache compatible. 

I ran the nlf (https://www.npmjs.com/package/nlf) tool on the package.json. This traverses the entire dependency tree for license information. It found primarily BSD and MIT. There were some ISC (http://en.wikipedia.org/wiki/ISC_license) that I couldn't find a direct reference to on the Apache site, but it looks open to me. There were a handful of modules that didn't specify a license, and one WTFPL (http://www.wtfpl.net/) which seems super compatible.

klucar commented:
I updated to the latest phase1 branch and resolved the conflicts. @DarinJ said he'll take a look and verify it runs soon.

DarinJ commented on setup-yarn-2.sh:
```diff
@@ -62,3 +63,4 @@ sudo -u hduser sh -c '/usr/local/hadoop/sbin/mr-jobhistory-daemon.sh start histo
 
 # Check status
 sudo -u hduser -u hduser jps
+
```
Why are we modifying this file? seems irrelevant to the dashboard.  If it needs fixed should it be a separate PR?

DarinJ commented on src/main/java/com/ebay/myriad/api/ClustersResource.java:
```diff
@@ -68,8 +67,15 @@ public Response flexUp(FlexUpClusterRequest request) {
         // TODO(mohit): Validation
+        LOGGER.info("Received Flexup Cluster Request");
+
+
```
Just one new line here.

DarinJ commented on src/main/resources/webapp/js/components/ClustersComponent.js:
```diff
@@ -0,0 +1,13 @@
+
+});
+
+module.exports = ClustersComponent;
```
I don't really get the purpose of the clusters page.  Is this something you need for future work?

kensipe replied:
I agree with @DarinJ.  I think I understand where this is going, however it is a potential future.   Currently we only support 1 cluster per framework.   unless I'm missing something... lets remove.

DarinJ commented:
A few in-line comments.  It would be nice to have a line in one of the main documentation pages mention the WebUI.  In a shippable state, should be merged.

DarinJ commented:
@kensipe were you going to look at this?  It's a blocker for a few other issues.

kensipe commented:
@DarinJ I fell behind on some work priorities... I'm on it now.   Did your comments get resolved?   specifically the comment about unrelated changes.  Has that been pulled out or will it be?

DarinJ commented:
@kensipe, Jim hasn't fixed the unrelated changes (setup-yarn-2.sh) though looking at the file I'd say its in a better state, but he can pull that out easy enough.

klucar commented:
just removed changes to setup-yarn-2.sh. Squashed all working commits into
one. Reflected issue #82 in commit message. pull request updated.

On Fri, May 29, 2015 at 2:05 PM, Darin notifications@github.com wrote:

> @kensipe https://github.com/kensipe, Jim hasn't fixed the unrelated
> changes (setup-yarn-2.sh) though looking at the file I'd say its in a
> better state, but he can pull that out easy enough.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mesos/myriad/pull/82#issuecomment-106892124.

kensipe commented on src/main/resources/webapp/README.md:
```diff
@@ -0,0 +1,27 @@
+
+A browser window should open with the site loaded. If not, it uses [port 8888](http://localhost:8888) 
+It is helpful to have myriad setup in Vagrant locally so the api is available.
+
```
I'm fine with this information being here... however it is an important part of the build process and a new developer should not have to hunt for this information.  there needs to be a comment and a link from the main project README to this file.

kensipe commented on src/main/resources/webapp/js/components/ConfigComponent.js:
```diff
@@ -0,0 +1,39 @@
+var RawJSONComponent = require('../components/RawJSONComponent');
+
+// localhost:8192/api/config
+//{"profiles":{"small":{"cpu":"1","mem":"1100"},"medium":{"cpu":"2","mem":"2048"},"large":{"cpu":"4","mem":"4096"}},"mesosMaster":"10.0.2.15:5050"}
```
why is this commented out?   can we just remove this?

kensipe commented on src/main/resources/webapp/js/components/Myriad.js:
```diff
@@ -0,0 +1,115 @@
+  componentWillUnmount: function() {
+
+  },
+*/
```
remove commented out code

kensipe commented on src/main/resources/webapp/js/components/ProfileComponent.js:
```diff
@@ -0,0 +1,25 @@
+var React = require('react');
+
+//{"small":{"cpu":"1","mem":"1100"}}
```
lets remove commented out code

kensipe commented on src/main/resources/webapp/js/components/TaskListComponent.js:
```diff
@@ -0,0 +1,20 @@
+
+  render: function () {
+
+    // props: name, taskNames
```
comment isn't meaningful... I don't see any value to the comment

kensipe commented:
@klucar I've made a few comments... lets get this merged today if possible.   Here is what is outstanding:
1. the setup-yarn-2.sh changes should be removed from this PR.  you reference in your last comment that this is taken care of, but I don't see it.
2. remove cluster stuff
3. add link to webapp/README in the main README (with comment)
4. remove commented out code from ConfigComponent.js, Myriad.js, ProfileComponent.js
   commented code:  TaskListComponent.js
5. misalignment of input text boxes on the flex tab

klucar commented:
The Clusters page is there only because the original UI had one. I couldn't
figure out what the original actually displayed, so this one is blank. I'll
gladly remove it, unless someone knows what it was meant for.

Most of the commented out stuff is just my development cruft so I can
remember what JSON the web service returns. I'll clean it up a bit, but the
props stuff is useful, so I may just clarify what they are better.

I should be able to do this later today.

Jim

On Mon, Jun 1, 2015 at 11:04 AM, Ken Sipe notifications@github.com wrote:

> @klucar https://github.com/klucar I've made a few comments... lets get
> this merged today if possible. Here is what is outstanding:
> 1. the setup-yarn-2.sh changes should be removed from this PR. you
>    reference in your last comment that this is taken care of, but I don't see
>    it.
> 2. remove cluster stuff
> 3. add link to webapp/README in the main README (with comment)
> 4. remove commented out code from ConfigComponent.js, Myriad.js,
>    ProfileComponent.js commented code: TaskListComponent.js
> 5. misalignment of input text boxes on the flex tab
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mesos/myriad/pull/82#issuecomment-107561927.

kensipe commented:
@klucar are you expecting this today?  I understand this is blocking some other work.

klucar commented:
I'm building and testing now.

How should I do the README link? (e.g. relative path, absolute github path,
something else)

On Tue, Jun 2, 2015 at 12:52 PM, Ken Sipe notifications@github.com wrote:

> @klucar https://github.com/klucar are you expecting this today? I
> understand this is blocking some other work.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mesos/myriad/pull/82#issuecomment-108013038.

kensipe commented:
I would use a relative link such that it works from github.

klucar commented:
what do you think about just moving it to the docs directory with
everything else? this makes more sense to me.

On Tue, Jun 2, 2015 at 4:20 PM, Ken Sipe notifications@github.com wrote:

> I would use a relative link such that it works from github.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mesos/myriad/pull/82#issuecomment-108086407.

klucar commented:
I moved the README.md to docs/myriad-dashboard.md

klucar commented:
Latest code should be good to go. Not sure why my setup-yarn-2.sh fix didn't take last time, but it looks like latest doesn't have any changes to setup-yarn-2.sh

kensipe commented:
LGTM!   Merging!
